src/bridge/cockpitinternalmetrics.c

```diff
@@ -402,20 +402,22 @@ convert_metric_description (CockpitInternalMetrics *self,
   MetricDescription *desc = find_metric_description (name);
   if (desc == NULL)
     {
-      g_warning ("%s: unknown internal metric %s", self->name, name);
-      return FALSE;
+      g_message ("%s: unknown internal metric %s", self->name, name);
     }
-
-  if (units && g_strcmp0 (desc->units, units) != 0)
+  else
     {
-      g_warning ("%s: %s has units %s, not %s", self->name, name, desc->units, units);
-      return FALSE;
-    }
+      if (units && g_strcmp0 (desc->units, units) != 0)
+        {
+          g_warning ("%s: %s has units %s, not %s", self->name, name, desc->units, units);
+          return FALSE;
+        }
+
+      if (desc->instanced)
+        info->instances = g_hash_table_new_full (g_str_hash, g_str_equal, g_free, g_free);
 
-  if (desc->instanced)
-    info->instances = g_hash_table_new_full (g_str_hash, g_str_equal, g_free, g_free);
+      info->desc = desc;
+    }
 
-  info->desc = desc;
   self->samplers |= desc->sampler;
```

Comment:
doesn't the code path `desc == NULL` lead to an invalid nullpointer dereference here? is that intentional?